src/ldp/operations/HeadOperationHandler.ts

```diff
@@ -0,0 +1,40 @@
+import { Readable } from 'stream';
+import type { ResourceStore } from '../../storage/ResourceStore';
+import { UnsupportedHttpError } from '../../util/errors/UnsupportedHttpError';
+import type { Operation } from './Operation';
+import { OperationHandler } from './OperationHandler';
+import type { ResponseDescription } from './ResponseDescription';
+
+/**
+ * Handles HEAD {@link Operation}s.
+ * Calls the getRepresentation function from a {@link ResourceStore}.
+ */
+export class HeadOperationHandler extends OperationHandler {
+  private readonly store: ResourceStore;
+
+  public constructor(store: ResourceStore) {
+    super();
+    this.store = store;
+  }
+
+  public async canHandle(input: Operation): Promise<void> {
+    if (input.method !== 'HEAD') {
+      throw new UnsupportedHttpError('This handler only supports HEAD operations.');
+    }
+  }
+
+  public async handle(input: Operation): Promise<ResponseDescription> {
+    const body = await this.store.getRepresentation(input.target, input.preferences);
+
+    // Close the Readable as we will not return it.
+    return Promise.resolve()
+      .then((): any => body.data.destroy())
+      .then((): any => {
+        body.data = new Readable();
+        body.data._read = function(): void {
+          body.data.push(null);
+        };
+        return { identifier: input.target, body };
+      });
```

Comment:
Wondering if the first Promise is necessary? I tried reproducing the error but it never seems to trigger for me so I can't really check (which is kind of unfortunate since that makes testing more difficult). But then it would be something like `return new Promise(resolve => { body.data.destroy(); resolve(); })...`. And if that was put in an async function you wouldn't need an explicit Promise object and could do something like `await destroyStream(); body.data = ...`.

Reply:
I'm somewhat confused because earlier the tests on different node versions failed because of this https://github.com/solid/community-server/pull/168#discussion_r497746920 but if I now unwrap it so there is no `new Promise` anymore, all tests pass. 
I'll push the unwrapping and we'll see if the Travis builds pass too.